cli/cmd/plugin/unmanaged-cluster/tanzu/tanzu.go

```diff
@@ -1025,10 +1057,83 @@ func mergeKubeconfigAndSetContext(mgr kubeconfig.Manager, kcPath string) error {
 	return nil
 }
 
+// resolveProfilePkg picks the first package in the package repo
+// that matches the name and version of the provided profile
+// If the user did not specify a profile version, defaults to the first one found which should be the latest
+func resolveProfilePkg(mgr packages.PackageManager, profileName, profileVersion string) (*Package, error) {
```

Comment:
Could we turn this into `resolvePkg` and re-use it for the resolveCNI logic too?

Seems like they are similar enough?

https://github.com/jpmcb/community-edition/blob/afe9592b4bbc4df82212f01432bb30ad1107d4d1/cli/cmd/plugin/unmanaged-cluster/tanzu/tanzu.go#L1136-L1160